internal/sipsource/config.go

```diff
@@ -0,0 +1,53 @@
+package sipsource
+
+import (
+	"errors"
+
+	"github.com/google/uuid"
+	"go.artefactual.dev/tools/bucket"
+)
+
+var (
+	// ErrMissingID is returned when the SIP source ID is missing.
+	ErrMissingID = errors.New("SIP source: missing ID")
+	// ErrMissingName is returned when the SIP source name is empty.
+	ErrMissingName = errors.New("SIP source: missing name")
+	// ErrMissingBucket is returned when no bucket is configured.
+	ErrMissingBucket = errors.New("SIP source: missing bucket")
+)
+
+type Config struct {
+	// ID is the UUID of the SIP source.
+	ID uuid.UUID
+
+	// Name is the human readable name of the SIP source.
+	Name string
+
+	// Bucket is the configuration for the bucket to be used as the SIP source.
+	Bucket *bucket.Config
+}
+
+func (c *Config) Validate() error {
+	// Allow empty SIP source configurations, for installations where no SIP
+	// source is needed.
+	if c.IsEmpty() {
+		return nil
+	}
+
+	var errs error
+	if c.ID == uuid.Nil {
+		errs = errors.Join(errs, ErrMissingID)
+	}
+	if c.Name == "" {
+		errs = errors.Join(errs, ErrMissingName)
+	}
+	if c.Bucket == nil {
+		errs = errors.Join(errs, ErrMissingBucket)
+	}
```

Comment:
We should validate the bucket config too, like we do with the the watched and the internal bucket.